pymc/gp/cov.py

```diff
@@ -194,23 +258,79 @@ def merge_factors(self, X, Xs=None, diag=False):
                     TensorSharedVariable,
                 ),
             ):
+
                 if factor.ndim == 2 and diag:
                     factor_list.append(at.diag(factor))
                 else:
                     factor_list.append(factor)
+
             else:
                 factor_list.append(factor)
+
+        return factor_list
+
+    def _merge_factors_psd(self, omega):
+        """Called to evaluatate spectral densities of combination kernels when possible.  Implements
+        a more restricted set of rules than `_merge_factors_cov` -- just additivity of stationary
+        covariances with defined power spectral densities and multiplication by scalars.  Also, the
+        active_dims for all covariances in the sum must be the same.
+        """
+        factor_list = []
+        for factor in self._factor_list:
+            if isinstance(factor, Covariance):
+
+                # Allow merging covariances for psd only if active_dims are the same
+                if set(self.active_dims) != set(factor.active_dims):
+                    raise ValueError(
+                        "For power spectral density calculations `active_dims` must be the same "
+                        "for all covariances in the sum."
+                    )
+
+                # If it's a covariance try to calculate the psd
+                try:
+                    factor_list.append(factor.psd(omega))
+
+                except (AttributeError, NotImplementedError) as e:
+
+                    if isinstance(factor, Stationary):
+                        raise NotImplementedError(
+                            f"No power spectral density method has been implemented for {factor}."
+                        ) from e
+
+                    else:
+                        raise ValueError(
+                            "Power spectral densities, `.psd(omega)`, can only be calculated for "
+                            f"`Stationary` covariance functions.  {factor} is non-stationary."
+                        ) from e
+
+            else:
+                # Otherwise defer the reduction to later
+                factor_list.append(factor)
+
         return factor_list
 
 
 class Add(Combination):
     def __call__(self, X, Xs=None, diag=False):
-        return reduce(add, self.merge_factors(X, Xs, diag))
+        return reduce(add, self._merge_factors_cov(X, Xs, diag))
+
+    def psd(self, omega):
```

Comment:
What is psd? Can you add type hints and a docstring?

Reply:
good call, changed to `power_spectral_density`